core/drivers/crypto/marvell/cn10k/include/ehsm-aes.h

```diff
@@ -0,0 +1,159 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (C) 2025 Marvell.
+ */
+
+#ifndef __EHSM_AES_H__
+#define __EHSM_AES_H__
+#include <stdint.h>
+#include <stdbool.h>
+
+#include "ehsm.h"
+#include "ehsm-security.h"
+
+enum ehsm_aes_mode {
+	EHSM_AES_MODE_ECB       = 0,
+	EHSM_AES_MODE_CBC       = 1,
+	EHSM_AES_MODE_CTR       = 2,
+	EHSM_AES_MODE_XTS       = 3,
+	EHSM_AES_MODE_KEY_WRAP  = 4,
+	EHSM_AES_MODE_CFB       = 5,
+	EHSM_AES_MODE_OFB       = 6,
+	EHSM_AES_MODE_GCM       = 7,
+};
+
+enum ehsm_aes_key_size {
+	EHSM_AES_KEY_128        = 128,
+	EHSM_AES_KEY_192        = 192,
+	EHSM_AES_KEY_256        = 256,
+};
+
+/**
```

Comment:
The extra `*` isn't needed since we don't generate documentation from this. These comments are only intended for humans.